moped-database/migrations/1699568640283_squashed-migration-add-audit-field-to-proj-components/up.sql

```diff
@@ -0,0 +1,22 @@
+alter table "public"."moped_proj_components" rename column "date_added" to "created_at";
+
+alter table "public"."moped_proj_components" add column "created_by_user_id" integer null;
+
+alter table "public"."moped_proj_components" add column "updated_by_user_id" integer null;
+
+alter table "public"."moped_proj_components" add column "updated_at" timestamptz null;
+
+CREATE OR REPLACE FUNCTION update_updated_at_column()
+RETURNS trigger
+LANGUAGE plpgsql
+AS $$ 
+BEGIN
+  NEW.updated_at = now();
+  RETURN NEW;
+END;
+$$;
```

Comment:
We'll see how it plays out, but my intention is to make this a generic function that works for any table so we can hook it up to whatever we want that has audit fields following our pattern.

Reply:
💯 this is great!

how would you feel about tearing down the replacing the existing function + trigger we have for the `moped_proj_work_activity` table in this PR?

Reply:
Yep, I'd love to. Will do.